easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1668,6 +1668,7 @@ def det_short_module_name(self, ec, force_visible=False):
         self.log.debug("Obtained valid short module name %s" % mod_name)
 
         # sanity check: obtained module name should pass the 'is_short_modname_for' check
+        ec = self.check_ec_type(ec)
```

Comment:
After thinking this through, I'm not sure that this does what you think/want it does...

`check_ec_type` will basically go out and ask the module naming scheme that is being used whether the provided `ec` dictionary is sufficient to query the module naming scheme.

This is used to determine whether or not a full easyconfig file is required for each dependency, or whether the limited information that is provided by the dependency specification (i.e., `name`, `version`, `versionsuffx` & `toolchain`) is sufficient (it is for the default `EasyBuildMNS`, not for `HierarchicalMNS` which also requires `moduleclass`).

So, unless the module naming scheme you are using specifies that `modaltsoftname` must also be provided, this probably doesn't have the effect you want it to have.

Since this is only a sanity check, we could limit it to the case where the MNS is being queried with a full `EasyConfig` instance, basically by only raising the error if `modaltsoftware` is indeed available.

What made you make this change exactly? Maybe it helps to have a bit more context here...